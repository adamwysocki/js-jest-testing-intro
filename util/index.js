/**
 * @function properCase
 * @description - function for proper casing names
 * @param {string} name - the name to be proper cased
 * @returns {string} - the proper cased name
 */
export const properCase = (name) => {
    if(!name) return name;
    return name.split(" ").map( (word) => {
        return word.charAt(0).toUpperCase() + word.slice(1);
    }).join(" ");
}