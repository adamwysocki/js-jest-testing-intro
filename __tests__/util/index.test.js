import { properCase } from '../../util';

/** test the properCase function */
describe('Unit tests for properCase util function', () => {
    it('should proper case a name', () => {
        const name = "adam wysocki";
        expect(properCase(name)).toBe("Adam Wysocki");
    });
    
    it('should proper case a name with a single word', () => {
        const name = "madonna";
        expect(properCase(name)).toBe("Madonna");
    });
    
    it('shoud proper case a name with more than two words', () => {
        const name = "john jacob jingleheimer schmidt";
        expect(properCase(name)).toBe("John Jacob Jingleheimer Schmidt");
    });
    
    it('should return null if passed null', () => {
        const name = null;
        expect(properCase(name)).toBeNull();
    });

    it('should match a snapshot', () => {
        const name = "miley cyrus";
        const properCasedName = properCase(name);
        expect(properCasedName).toMatchSnapshot();
    });
});
