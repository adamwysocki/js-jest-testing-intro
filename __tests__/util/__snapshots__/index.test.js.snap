// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for properCase util function should match a snapshot 1`] = `"Miley Cyrus"`;
